pkg/agent/config.go

```diff
@@ -303,30 +336,44 @@ func InitAgentCmdFlags(c *cobra.Command, cfg *AgentCmdFlags) {
 	)
 	c.PersistentFlags().MarkDeprecated("disable-compression", "no longer has an effect")
 
+	c.PersistentFlags().BoolVar(
+		&cfg.MachineHubMode,
+		"machine-hub",
+		false,
+		"Enables MachineHub mode. The agent will push data to CyberArk MachineHub. Can be used in conjunction with --venafi-cloud.",
+	)
```

Comment:
suggestion: I agree that this is probably not required in the long term, we can just enable it if configuration is provided.

In the short term, I think this feature will be rolled out slowly and we'll want it to be there for testing before it's ready for production.

With that in mind, I think it makes sense to have this flag, but hidden by default or else with a description that indicates the feature is for testing only. What do you think?

Reply:
I like the idea of hiding it so that we can roll this out without waiting. I'll do that.

Reply:
Done. Here is how I describe `--machine-hub`:

> This is a hidden feature flag we use to build the "Machine Hub" feature gradually without impacting customers. Once the feature is GA, we will turn this flag "on" by default.